### ISSUE TITLE
Options for customizing how racket-xp-mode displays info text?

### ISSUE THREAD

**Issue by jtherrmann** [greghendershott/racket-mode#495]
Thanks for your great work on racket-mode. My only point of friction is that I find the info text in `racket-xp-mode` a bit distracting. The [docs](https://www.racket-mode.com/#racket_002dxp_002dmode) explain how to disable `racket-xp-pre-redisplay`, but then it seems the only way to trigger the info text is by mousing over a particular annotation, and I would prefer to keep my hands on the keyboard. I'm wondering if it's possible to instead configure `racket-xp-mode` to do any of the following:

* Display the info text in the minibuffer
* Display the info text near the bottom of the screen, with light text on a dark background
* Display the info text only when a particular command is run (I don't want to run `racket-xp-annotate` manually, I just want the info text hidden until I request it)

I think that any of these options would help reduce the distraction.

**Comment by greghendershott** [comment]
Have you tried changing the customization variable [`racket-show-functions`](https://www.racket-mode.com/#racket_002dshow_002dfunctions) from its default value of [`'(racket-show-pseudo-tooltip)`](https://www.racket-mode.com/#racket_002dshow_002dpseudo_002dtooltip) to say [`'(racket-show-echo-area)`](https://www.racket-mode.com/#racket_002dshow_002decho_002darea)? That sounds like your first bullet point?

Note that it's `racket-show-functions` plural, the value is a list, so you can also try enabling multiple methods at once, and later remove the ones that annoy you. Also if none are precisely what you want, you could supply your own function; I could try to help you with that.

**Comment by greghendershott** [comment]
Also the [`racket-show-header-line`](https://www.racket-mode.com/#racket_002dshow_002dheader_002dline) function sounds like your second bullet, albeit top of the buffer not bottom. (The Emacs "header line" is like the mode line, for a buffer. For most modes it's blank/invisible, but it's an option to use when you don't want to clobber the mode line or echo area.)

**Comment by jtherrmann** [comment]
Ah, yes, one of those will definitely do the job. Sorry for not reading the docs more carefully -- I did scan all of the variables at https://www.racket-mode.com/ but I missed `racket-show-functions`. Thanks for the help!

**Comment by jtherrmann** [comment]
@greghendershott I decided that I still wanted the ability to hide the info text unless I trigger it with a command. I did not like the idea of running `racket-xp-annotate` manually, because it does not automatically clear the info text after point leaves the related symbol. This is what I came up with instead:

```elisp
(defun jth-racket-show (val &optional pos)
  (setq jth-racket-saved-info val))

(setq racket-show-functions '(jth-racket-show))

(defun jth-racket-display-info ()
  (interactive)
  (message "%s" jth-racket-saved-info))

(add-hook 'racket-mode-hook
	  (lambda () (local-set-key (kbd "C-c r") 'jth-racket-display-info)))
```

This solution automatically saves the info text in `jth-racket-saved-info` until I print it with the `jth-racket-display-info` command.

Let me know if there was a builtin way to do this that I missed. If not, perhaps this would be a useful feature for other users? I could make a PR with a more flexible solution (i.e. the ability to show the info text with the functions in `racket-show-functions` instead of just printing to the echo area) if you think it would be worthwhile. No worries if not, just thought I'd make the suggestion.
